packages/web3-errors/src/errors/response_errors.ts

```diff
@@ -15,11 +15,23 @@ You should have received a copy of the GNU Lesser General Public License
 along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
 */
 
-/* eslint-disable max-classes-per-file */
+// eslint-disable-next-line max-classes-per-file
+import { JsonRpcResponse, JsonRpcResponseWithError } from 'web3-types';
+import { Web3Error } from '../web3_error_base';
+import { ERR_INVALID_RESPONSE, ERR_RESPONSE } from '../error_codes';
 
-import { ERR_INVALID_RESPONSE, ERR_RESPONSE, Web3Error } from 'web3-errors';
-import { isResponseWithError } from './json_rpc';
-import { JsonRpcResponse } from './types';
+// To avoid circular package dependency, copied to code here. If you update this please update same function in `json_rpc.ts`
```

Comment:
For a large code base in JS ecosystem, number of dependencies is not that valuable over a maintainable codebase. That's why the concept of the monorepo became so much popular, as it allows you to create domain specific application components in different packages to have clear understanding. 

Otherwise I would assume over 90% of the users are just concerned to `web3` package, so why not just create one with zero interdependencies.